presto-spi/src/main/java/com/facebook/presto/spi/session/SessionPropertyRule.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.session;
+
+import java.util.Map;
+
+public interface SessionPropertyRule
+{
+    int getPriority();
```

Comment:
If multiple rules apply and have conflicts, who wins?

Reply:
Let's remove this. It's confusing because the method `getRulesInPriorityOrderLowToHigh()` already returned the ordered list. Also, it's not used by the engine. Implementations are free to store a priority internally if they want one.

Reply:
The `getRulesInPriorityOrderLowToHigh` method uses `getPriority` to sort the rules. I can't see how to remove this.

Reply:
The implementation of the class is free to have a priority field or whatever it wants. But this doesn't need to be in the interface, since it's only needed by the plugin to sort the list for the `getRulesInPriorityOrderLowToHigh()` method.